registries/adapters/apiv2_adapter.go

```diff
@@ -0,0 +1,249 @@
+//
+// Copyright (c) 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package adapters
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/automationbroker/bundle-lib/bundle"
+	"github.com/automationbroker/bundle-lib/registries/adapters/oauth"
+	log "github.com/sirupsen/logrus"
+)
+
+const apiV2ManifestURL = "%v/v2/%v/manifests/%v"
+const apiV2CatalogURL = "%v/v2/_catalog"
+
+// OpenShiftAdapter - OpenShift Adapter
+type OpenShiftAdapter struct {
+	apiV2Adapter
+}
+
+// PartnerRhccAdapter - Partner RHCC Adapter
+type PartnerRhccAdapter struct {
```

Comment:
Well mostly because I whined about it. I preferred having the actual types vs just using a generic type. Probably for preoptimization, but I suspect at some point you'll need to fix something for PartnerRhccAdapter that isn't supposed to be fixed in DockerHub. But to me from a maintenance point of view. If a bug comes in stating they're having trouble connecting to PartnerRhcc, I know EXACTLY where to look. Vs looking for the config to find the magic string, then finding out it's APIV2, then find APIV2. Also, it's almost a form of inheritance (poor choice of word for a language that doesn't have it, but still kinda is), PartnerRhcc AND DockerHub *are* APIV2 Adapters.

Reply:
@jmrodri, would you object to exporting `apiV2Adapter` and add a `v2adapter` type to `registry.go`?

Reply:
@johnkim76 whatever, I've said my peace, expose APIV2. @dymurray as a user I know I'm trying to connect to openshift registry or partner registry. I don't really *know* they're APIV2. But we've gone on long enough, you guys have spent more time in registry stuff.